src/engine/Collision/Body.ts

```diff
@@ -108,139 +124,184 @@ export class Body {
+   * Sets the old versions of pos, vel, acc, and scale.
    */
-  public getRelativeBounds(): BoundingBox {
-    if (Physics.collisionResolutionStrategy === CollisionResolutionStrategy.Box) {
-      return this.actor.getRelativeBounds();
-    } else {
-      return this.actor.getRelativeBounds();
-    }
+  public captureOldTransform() {
+    // Capture old values before integration step updates them
+    this.oldVel.setTo(this.vel.x, this.vel.y);
+    this.oldPos.setTo(this.pos.x, this.pos.y);
+    this.oldAcc.setTo(this.acc.x, this.acc.y);
+    this.oldScale.setTo(this.scale.x, this.scale.y);
   }
 
   /**
-   * Updates the collision area geometry and internal caches
+   * Perform euler integration at the specified time step
    */
-  public update() {
-    if (this.collisionArea) {
-      // Update the geometry if needed
-      if (this.actor && this.actor.isGeometryDirty && this.collisionArea instanceof PolygonArea) {
-        this.collisionArea.points = this.actor.getRelativeGeometry();
-      }
+  public integrate(delta: number) {
+    // Update placements based on linear algebra
+    const seconds = delta / 1000;
+
+    const totalAcc = this.acc.clone();
+    // Only active vanilla actors are affected by global acceleration
+    if (this.collider.type === CollisionType.Active) {
+      totalAcc.addEqual(Physics.acc);
+    }
+
+    this.vel.addEqual(totalAcc.scale(seconds));
+    this.pos.addEqual(this.vel.scale(seconds)).addEqual(totalAcc.scale(0.5 * seconds * seconds));
+
+    this.rx += this.torque * (1.0 / this.collider.inertia) * seconds;
+    this.rotation += this.rx * seconds;
+
+    this.scale.x += (this.sx * delta) / 1000;
+    this.scale.y += (this.sy * delta) / 1000;
 
-      this.collisionArea.recalc();
+    if (!this.scale.equals(this.oldScale)) {
+      // change in scale effects the geometry
+      this._geometryDirty = true;
     }
+
+    // Update colliders
+    this.collider.update();
+    this._geometryDirty = false;
   }
 
   /**
-   * Sets up a box collision area based on the current bounds of the associated actor of this physics body.
+   * Sets up a box geometry based on the current bounds of the associated actor of this physics body.
    *
    * By default, the box is center is at (0, 0) which means it is centered around the actors anchor.
    */
-  public useBoxCollision(center: Vector = Vector.Zero) {
-    this.collisionArea = new PolygonArea({
-      body: this,
-      points: this.actor.getRelativeGeometry(),
+  public useBoxCollider(center: Vector = Vector.Zero): Collider {
+    this.collider.shape = new ConvexPolygon({
+      collider: this.collider,
+      points: this._actor.getRelativeGeometry(),
       pos: center // position relative to actor
     });
 
     // in case of a nan moi, coalesce to a safe default
-    this.moi = this.collisionArea.getMomentOfInertia() || this.moi;
+    this.collider.inertia = this.collider.shape.getInertia() || this.collider.inertia;
```

Comment:
This seems odd, should this ever be NaN? Or should we have a better isNaN() check around this?